packages/flutter_tools/lib/src/application_package.dart

```diff
@@ -161,7 +161,13 @@ class AndroidApk extends ApplicationPackage {
       return null;
 
     final String manifestString = manifest.readAsStringSync();
-    final xml.XmlDocument document = xml.parse(manifestString);
+    xml.XmlDocument document;
+    try {
+      document = xml.parse(manifestString);
+    } on Exception {
+      printError('AndroidManifest.xml is not a valid XML document.');
+      return null;
```

Comment:
We may want `throwToolExit` here - I had been hoping we could try to tell them what is wrong with the XML document, but that may not be possible due to the variety of potential errors.  At any rate, it would be an improvement to be telling them that it's the `AndroidManifest.xml` - maybe telling them the relative path to it from the root of the project as well.